plugins/module_utils/foreman_helper.py

```diff
@@ -211,6 +211,35 @@ def find_resources_by_name(self, resource, names, **kwargs):
     def find_resources_by_title(self, resource, titles, **kwargs):
         return [self.find_resource_by_title(resource, title, **kwargs) for title in titles]
 
+    def find_operatingsystem(self, name, params=None, failsafe=False, thin=None):
```

Comment:
So I still think that this is more readable, while taking two API calls:

```python
def find_operatingsystem(self, name, params=None, failsafe=False, thin=None):
  operatingsystem = self.find_resource_by_title('operatingsystems', name, params=params, failsafe=True, thin=thin)
  if not operatingsystem:
    search = 'title~"{}"'.format(name)
    operatingsystem = self.find_resource('operatingsystems', search, params=params, failsafe=failsafe, thin=thin)
  return operatingsystem
```

But I wouldn't block on that.

Reply:
I prefer to do less api calls. (They are expensive)

Reply:
I'd disagree that they are expensive for *operatingsystems*. Hosts yes. But OS...
We're just adding a single `index` call (because if that succeeds, the second find is not executed, and if it does not succeed, no further API calls inside `find_resource` are executed).

Anyways. I said I won't block on this, and I stand by that. :)